cirq/sim/density_matrix_simulator.py

```diff
@@ -276,7 +275,8 @@ def keep(potential_op: ops.Operation) -> bool:
             for op in channel_ops_and_measurements:
                 indices = [qubit_map[qubit] for qubit in op.qubits]
                 # TODO: support more general measurements.
-                meas = ops.op_gate_of_type(op, ops.MeasurementGate)
+                meas = op.gate if isinstance(op.gate,
+                                             ops.MeasurementGate) else None
                 if meas:
```

Comment:
```
if isinstance(op.gate, ops.MeasurementGate):
    meas = op.gate
```

If we could use python 3.8:
```
if isinstance(meas := op.gate, ops.MeasurementGate):
```

Reply:
yup; I'm very excited for the walrus operator :-)